AGENTS.md

```diff
@@ -0,0 +1,49 @@
+This is the Kubernetes project, also known as K8s. It is an open-source container orchestrator released under the Apache 2 license, designed to run and manage workloads at scale on major cloud providers and on-premises.
+
+You are an expert AI programming assistant with a specialization in the Go implementation of Kubernetes.
+
+### Project Conventions and Workflow
+
+To contribute effectively, you must adhere to the following project conventions:
+
+-   **Repository Structure**: The repository is organized into several key directories:
+    -   `cmd/`: Contains the `main` packages for the core Kubernetes binaries (kube-apiserver, kube-controller-manager, etc.).
+    -   `pkg/`: Contains the internal packages that implement the logic for the binaries in `cmd/`.
+    -   `staging/`: Contains code that is developed in this repository but published as separate `k8s.io` modules. Changes to these modules must be made here.
+    -   `hack/`: Contains essential build, test, and code generation scripts. You will use these scripts to ensure changes meet project standards.
+    -   `test/`: Contains end-to-end (e2e) tests for the project.
+
+-   **Code Generation**: Kubernetes heavily relies on code generation for API types, clientsets, informers, and more.
+    -   After modifying API definitions (in `k8s.io/api` or other API group directories), you **must** run `hack/update-codegen.sh` to regenerate the necessary files.
+
+-   **Verification and Formatting**:
+    -   To format your code, run `hack/update-gofmt.sh`.
```

Comment:
Most of our docs point to various make targets. This should be `make verify WHAG="gofmt". Maybe worth using that instead of directly `hack/*` scripts. Same applies to `hack/verify-all.sh` below. Just use `make verify`.